src/sentry/static/sentry/app/components/hoverCard.jsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+
+import Avatar from './avatar';
+
+import LoadingIndicator from './loadingIndicator';
+import LoadingError from './loadingError';
+
+import ApiMixin from '../mixins/apiMixin';
+
+const HoverCard = React.createClass({
+  propTypes: {
+    version: React.PropTypes.string.isRequired,
+    orgId: React.PropTypes.string.isRequired,
+    projectId: React.PropTypes.string.isRequired,
+    visible: React.PropTypes.bool.isRequired,
+  },
+
+  mixins: [ApiMixin],
+
+  getInitialState() {
+    return {
+      loading: true,
+      error: false,
+      data: {},
+    };
+  },
+
+  componentDidMount() {
+    let {orgId, version} = this.props;
+
+    let path = `/organizations/${orgId}/releases/${version}/`;
+    this.api.request(path, {
+      method: 'GET',
+      success: (data, _, jqXHR) => {
+        this.setState({
+          release: data,
+          loading: false,
+        });
+      },
+      error: () => {
+        this.setState({
+          error: true,
+        });
+      }
+    });
+  },
+
+  render() {
+
+    let {version, visible} = this.props;
+    let shortVersion = version.match(/^[a-f0-9]{40}$/) ? version.substr(0, 12) : version;
```

Comment:
This should really be a utility function.

Also – we should verify the modal works with long version strings, since only shortens 40 character lower-case hex strings.